src/main/java/org/tugraz/sysds/runtime/instructions/cp/UnaryCPInstruction.java

```diff
@@ -76,6 +76,9 @@ public static UnaryCPInstruction parseInstruction ( String str ) {
 			else if(in.getDataType() == DataType.MATRIX)
 				return new UnaryMatrixCPInstruction(LibCommonsMath.isSupportedUnaryOperation(opcode) ?
 					null : InstructionUtils.parseUnaryOperator(opcode), in, out, opcode, str);
+			else if(in.getDataType() == DataType.FRAME)
+				return new UnaryFrameCPInstruction(InstructionUtils.parseUnaryOperator(opcode), in, out, opcode, str);
+			//todo return frame
```

Comment:
First, use TODO instead of todo to indicate open issues. Second, why would you want to return a frame instead of an instruction here?